usb.c

```diff
@@ -161,10 +161,13 @@ static int qdl_try_open(libusb_device *dev, struct qdl_device *qdl, const char *
 	return !!qdl->usb_handle;
 }
 
-int qdl_open(struct qdl_device *qdl, const char *serial)
+
+
+static int usb_open(struct qdl_device *qdl, const char *serial)
 {
 	struct libusb_device **devs;
 	struct libusb_device *dev;
+	struct qdl_device_usb *qdl_usb = (struct qdl_device_usb*) qdl;
```

Comment:
Please introduce and use a container_of() macro for this, rather than just casting between the types.

Reply:
added, thanks!